lib/unittest.bzl

```diff
@@ -412,6 +429,31 @@ def _expect_failure(env, expected_failure_msg = ""):
     else:
         _fail(env, "Expected failure of target_under_test, but found success")
 
+def _target_actions(env):
+    """Returns a list of actions registered by the target under test.
+
+    Args:
+      env: The test environment returned by `analysistest.begin`.
+    """
+
+    # Validate?
+    dep = _target_under_test(env)
+    return dep[_ActionInfo].actions
+
+def _target_under_test(env):
+    """Returns the target under test.
+
+    Args:
+      env: The test environment returned by `analysistest.begin`.
+    """
+    result = getattr(env.ctx.attr, "target_under_test")
+    if type(result) == type([]):
```

Comment:
types.bzl - `types.is_list(result)`

Reply:
Nice. I should really get used to using the libraries already present in this repository ;)